src/complex/DataStructure/DataStructure.cpp

```diff
@@ -946,13 +951,13 @@ void DataStructure::recurseHeirarchyToText(std::ostream& outputStream, const std
 
     // pull child paths or skip to next iteration
     auto optionalChildPaths = GetAllChildDataPaths(*this, path);
-    if(!optionalChildPaths.has_value() || optionalChildPaths.value().size() == 0)
+    if(!optionalChildPaths.has_value() || optionalChildPaths.value().empty())
     {
       continue;
     }
 
     // recurse
```

Comment:
```suggestion
    if(optionalDataPaths.has_value() && !optionalDataPaths.value().empty())
    {
      recurseHierarchyToText(outputStream, optionalChildPaths.value(), indent);
    }
```
repair the bug in this function as well